at_commons/lib/src/keystore/at_key.dart

```diff
@@ -13,6 +14,71 @@ class AtKey {
     return 'AtKey{key: $key, sharedWith: $sharedWith, sharedBy: $sharedBy, namespace: $namespace, metadata: $metadata, isRef: $isRef}';
   }
 
+  ///Builds a public key and returns a [PublicKeyBuilder]
+  ///
+  ///Example: public:phone.wavi@alice.
+  ///```dart
+  ///AtKey publicKey = AtKey.public('phone', 'wavi').build();
+  ///```
+  static PublicKeyBuilder public(String key, String namespace,
+      {ValueType valueType = ValueType.text}) {
+    return PublicKeyBuilder(valueType)
+      ..key(key)
+      ..namespace(namespace);
+  }
+
+  ///Builds a sharedWith key and returns a [SharedKeyBuilder]. Optionally the key
+  ///can be cached on the [AtKey.sharedWith] atSign.
+  ///
+  ///Example: @bob:phone.wavi@alice.
+  ///```dart
+  ///AtKey sharedKey = (AtKey.shared('phone', 'wavi')
+  ///     ..sharedWith('@bob')).build();
+  ///```
+  /// To cache a key on the @bob atSign.
+  /// ```dart
+  ///AtKey atKey = (AtKey.shared('phone', 'wavi')
+  ///  ..sharedWith('bob')
+  ///  ..cache(1000, true))
+  ///  .build();
+  /// ```
+  static SharedKeyBuilder shared(String key, String namespace,
+      {ValueType valueType = ValueType.text}) {
+    return SharedKeyBuilder(valueType)
+      ..key(key)
+      ..namespace(namespace);
+  }
+
+  /// Builds a self key and returns a [SelfKeyBuilder].
+  ///
+  /// Self keys are the private key's which are only accessed by the owner of the
```

Comment:
Self keys are private keys which can be accessed only by the owner of the atsign